src/main/java/com/marginallyclever/donatello/nodes/LoadTurtle.java

```diff
@@ -0,0 +1,47 @@
+package com.marginallyClever.donatello.nodes;
+
+import com.marginallyClever.makelangelo.makeArt.io.vector.TurtleFactory;
+import com.marginallyClever.makelangelo.turtle.Turtle;
+import com.marginallyClever.nodeGraphCore.Node;
+import com.marginallyClever.nodeGraphCore.NodeVariable;
+
+import java.awt.geom.Rectangle2D;
+
+public class LoadTurtle extends Node {
+    private final NodeVariable<String> filename = NodeVariable.newInstance("filename",String.class,null,true,false);
+    private final NodeVariable<Turtle> contents = NodeVariable.newInstance("contents", Turtle.class, new Turtle(),false,true);
+    private final NodeVariable<Number> w = NodeVariable.newInstance("width", Number.class, 0,false,true);
+    private final NodeVariable<Number> h = NodeVariable.newInstance("height", Number.class, 0,false,true);
+
+    public LoadTurtle() {
+        super("LoadTurtle");
+        addVariable(filename);
+        addVariable(contents);
+        addVariable(w);
+        addVariable(h);
+    }
+
+    public LoadTurtle(String filename) {
+        this();
+        this.filename.setValue(filename);
+    }
+
+    @Override
+    public Node create() {
+        return new LoadTurtle();
```

Comment:
this method is constructed in the same way in each class, have you tried to transform `Node` to a generic type and let this class instantiate it like :  `public class LoadTurtle extends Node<LoadTurtle>  {`

Reply:
how would the addVariable() get called?

Reply:
where do you have this ?